extensions/amazon-lambda-resteasy/runtime/src/main/java/io/quarkus/amazon/lambda/resteasy/runtime/container/ResteasyLambdaContainerHandler.java

```diff
@@ -71,15 +88,84 @@ protected AwsHttpServletResponse getContainerResponse(AwsProxyHttpServletRequest
     }
 
     @Override
-    protected void handleRequest(AwsProxyHttpServletRequest httpServletRequest, AwsHttpServletResponse httpServletResponse,
-            Context lambdaContext)
-            throws Exception {
-        Timer.start("RESTEASY_HANDLE_REQUEST");
+    protected void handleRequest(AwsProxyHttpServletRequest request, AwsHttpServletResponse response, Context context) {
+        VirtualClientConnection connection = VirtualClientConnection.connect(VertxHttpRecorder.VIRTUAL_HTTP);
+        try {
+            nettyDispatch(connection, request, response);
+        } catch (Exception e) {
+            response.setStatus(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            connection.close();
+        }
+    }
+
+    private void nettyDispatch(VirtualClientConnection connection, AwsProxyHttpServletRequest request,
+            AwsHttpServletResponse response) throws Exception {
+        String path = request.getPathInfo();
 
-        httpServletRequest.setServletContext(getServletContext());
+        String query = request.getQueryString();
+        if (query != null) {
+            path = path + '?' + query;
+        }
+
+        String host = request.getRemoteHost();
+        if (request.getRemotePort() != -1) {
+            host = host + ':' + request.getRemotePort();
+        }
 
-        doFilter(httpServletRequest, httpServletResponse, null);
-        Timer.stop("RESTEASY_HANDLE_REQUEST");
+        DefaultHttpRequest nettyRequest = new DefaultHttpRequest(HttpVersion.HTTP_1_1,
+                HttpMethod.valueOf(request.getMethod()), path);
+        nettyRequest.headers().set("Host", host);
+        for (Map.Entry<String, List<String>> header : request.getAwsProxyRequest().getMultiValueHeaders().entrySet()) {
+            nettyRequest.headers().add(header.getKey(), header.getValue());
+        }
+
+        HttpContent requestContent = LastHttpContent.EMPTY_LAST_CONTENT;
+        if (request.getAwsProxyRequest().getBody() != null) {
+            String body = request.getAwsProxyRequest().getBody();
+            ByteBuf buffer = Unpooled.wrappedBuffer(body.getBytes());
```

Comment:
This is not correct, in addition to not specifying the Charset when calling getBytes() in many cases the body may be base64 encoded. You need to call com.amazonaws.serverless.proxy.model.AwsProxyRequest#isBase64Encoded and use that to figure out if the request is encoded, and if so use the base 64 encoded to decode it. If it is not encoded you need to use the correct charset when calling getBytes, I think this should respect the Content-Type header of the request, and if it is not present default to UTF-8 (but that is just a guess, probably best to check the docs). 

It is also faster to call io.netty.buffer.Unpooled#copiedBuffer(java.lang.CharSequence, java.nio.charset.Charset) and let netty take care of the decoding.